### ISSUE TITLE
inprove soccer field grass apperence (ideas inside)

### ISSUE THREAD

**Issue by tobyplowy** [supertuxkart/stk-code#3053]
first off sorry for my bad spelling i did my best.

the grass in soccer field looks very flat (becouse it is) and thats a real shame becouse its a very inportant thing in this map (you see it everywhere on the map)

methode 1 simple doesent look great with all textures but its lowpolly 
(note how the sylinder sinks in the grass) 
![naamloos](https://user-images.githubusercontent.com/16853304/32996454-5b5070c2-cd83-11e7-927c-1df06a5a2059.png)
its made with layers. this can be done with images like i did here but it can also be done with postprossesing that usses only one texture
note i dout own these textures (i got the original fom google images i just use it for demo porpossus)
![racetrack_grass - kopie - kopie - kopie](https://user-images.githubusercontent.com/16853304/32996561-8b5fecec-cd84-11e7-999d-553b57c7e219.png)
![racetrack_grass - kopie - kopie](https://user-images.githubusercontent.com/16853304/32996562-8b777204-cd84-11e7-9887-fd627fba277c.png)
![racetrack_grass - kopie](https://user-images.githubusercontent.com/16853304/32996563-8b91651a-cd84-11e7-8e10-dec3270df1c8.png)
![racetrack_grass](https://user-images.githubusercontent.com/16853304/32996564-8baa47ec-cd84-11e7-9fe6-8c920a2b55ca.jpg)

methode 2 more polly count better visuals
agayn i dout own this image i use it for demo reasons
![cattail-grass-texture-3d-model-low-poly-max-obj-fbx](https://user-images.githubusercontent.com/16853304/32996576-b547351a-cd84-11e7-9335-4803e7ad1268.jpg)
this is how rocket league does its grass ist made using grass objects this can probabbly also be done with particles (note how the car sinks in the grass)
![rocket league pigeon](https://user-images.githubusercontent.com/16853304/32996611-3200f71c-cd85-11e7-824b-e5b171107ac9.jpg)

i hope it was usefull or at least inspiering
i hope i explaned correctly and that you guys can understand what i typed above agayn sorry for the spelling mistakes

**Comment by tobyplowy** [comment]
rocket league grass a better look
![lfpst2n](https://user-images.githubusercontent.com/16853304/32996688-32c48f00-cd86-11e7-9b3a-8568b8f6b485.png)

**Comment by tobyplowy** [comment]
@hiker I hope my spelling wasn't too bad (I dident have outo correct)
So is there any news on this is it a good idea?

**Comment by auriamg** [comment]
@tobyplowy This might be done if some artist is willing to play with this. We depend on contributions, artists will do what they want

**Comment by hiker** [comment]
Well, I don't think any one of the core team will have time for that, so unless we get a pull request, it will in the best case be a long time before we might consider it. If someone from the core team should be interested, they will just change the label and assign it to themselves, so you should get a notification in that instance.

**Comment by samuncle** [comment]
Hi
I'm sam the lead artist of SuperTuxKart and I'm interested however you can't just take an example from outside and except it to work in our game.

I took actually the time to test both solutions

1) The first one is using a technique that was developed for the shadow of the colossus and the result is surprisingly good in SuperTuxKart.

However it requires custom shaders and currently it's not easy to integrate custom shaders in game.
The test was done in a branch which allow easy integration of shaders however it's still Work In Progress and there is no guarantee it will be included in the core game.

I didn't noticed any performance issues. Here is a screenshot
![grass_shader](https://user-images.githubusercontent.com/6406045/33800293-3c552130-dd3d-11e7-8546-7b70f3efc135.jpg)

2) The second technique is already used in game however not with that much density. During my test the result wasn't impressive. Everything was quite dark and it require a high polycount which might be a bit unrealistic for the performances of the game.

![grass_shader_2](https://user-images.githubusercontent.com/6406045/33800484-99b14b94-dd40-11e7-84d5-f14fc557831f.jpg)

I'm not sure if it's the right approach for stk.

I would suggest probably a combination of both and the first technique can be applied for very specify cases like a stadium or a surface that is mainly composed of grass.

3) Since I was trying high density grass technique, I also made a test with billboards. The result is an interesting crossover. It has a lower poly count and produce less artifact than the 2. However the density wasn't good. It would be probably used in situation like small bushes in a desert.

![grass_shader_3](https://user-images.githubusercontent.com/6406045/33800593-3f91dc70-dd43-11e7-9d28-9e039e1dc7f6.jpg)

In conclusion the first method is quite interesting and if you could provide me with grass textures (as the one you included but creative commons or public domain) it would be a good start.

For a more generic approach I think the billboard is probably where we will go in the future.

**Comment by tobyplowy** [comment]
@samuncle thankyou for having a look in to my idea i will look for a texture

**Comment by tobyplowy** [comment]
@samuncle what about these?
https://opengameart.org/content/synthetic-grass-texture-pack
im not sure if the license is good (i dount understand anyting about licenses)
it uses CC0 1.0 Universal (CC0 1.0) 
in that pack you can find this texture (i think its pefect for the job)
![ground_grass_gen_05](https://user-images.githubusercontent.com/16853304/33800818-0fb4082e-dd49-11e7-8260-408b3d8ebaff.png)
note that the above image is a downscale

**Comment by tobyplowy** [comment]
@samuncle but i think that billboards wuld be best

**Comment by samuncle** [comment]
@tobyplowy we would need good alpha grass textures like the one you have shown in your post (to make the shadow of colossus technique).

I have no idea how to make them or generate them.

**Comment by tobyplowy** [comment]
@samuncle i made my own textures i hope they are good 
(my camera is kind of crap but it shuld do the trick)
what it shuld look like (note the cube and cone)
![grass](https://user-images.githubusercontent.com/16853304/33806069-9b942e4a-ddc2-11e7-8041-486dfee8e1c0.png)

the copyright is free to use i for what ever you whant creddit is apreciated :D
![1](https://user-images.githubusercontent.com/16853304/33806036-302c7932-ddc2-11e7-80c4-f37f2eee5161.png)
![2](https://user-images.githubusercontent.com/16853304/33806037-3041f906-ddc2-11e7-85db-33b44314865b.png)
![3](https://user-images.githubusercontent.com/16853304/33806038-3058e40e-ddc2-11e7-8070-0941abebbd64.png)
![4](https://user-images.githubusercontent.com/16853304/33806039-306df0ec-ddc2-11e7-8e27-088414431740.png)
![5](https://user-images.githubusercontent.com/16853304/33806040-30864f98-ddc2-11e7-85ba-d98b45027534.png)

**Comment by tobyplowy** [comment]
@samuncle is that texture good?

**Comment by Alayan-stk-2** [comment]
That certainly looks nicer than most grass in STK tracks.

**Comment by tobyplowy** [comment]
@Alayan-stk-2 yeah it does :)

**Comment by tobyplowy** [comment]
Is this still relevant?

**Comment by qwertychouskie** [comment]
STK now supports custom shaders.  @samuncle can you take a look at this again?  Especially since networked multiplayer will soon be in beta, soccer fields will probably see a lot more use.  If you provide a working example, I can adapt the existing tracks.

**Comment by tobyplowy** [comment]
It would be amazing if all the grass in every track had this feature :)

**Comment by tobyplowy** [comment]
@auriamg I'm possibly interested in helping with art stuff.
How do I get involved? I'm a texture/modeling artist but my main interest is texturing.

**Comment by auriamg** [comment]
If you want to get involved, check the "Get involved" section on our website https://supertuxkart.net/Community . Start by installing the blender tools, reading the documentation on how to make STK tracks. Probably start by making add-on tracks and/or karts to get some practice.
